### ISSUE TITLE
SyntaxError on python2.7.5 in 0.15.3

### ISSUE THREAD

**Issue by golf-player** [pallets/werkzeug#1544]
Traceback:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python2.7/site-packages/werkzeug/routing.py", line 948
    exec(code, globs, locs)
SyntaxError: unqualified exec is not allowed in function '_compile_builder' it contains a nested function with free variables
```

I haven't dug too deep into this, but I suspect it's the comprehensions which, in python2.7.5, are treated as subfunctions https://stackoverflow.com/a/52498591/8826764

python2.7.5 is important to support as it's the default python version shipped with centos7, and we're not at EOL quite yet.

I'll have time on the weekend to make an MR for this if that's OK with y'all.

If not, please change the setup python_requires to require a higher version of python.

**Comment by davidism** [comment]
@asottile

**Comment by asottile** [comment]
should be an easy fix, let me see what I can do here, I've never seen this error before though!

**Comment by golf-player** [comment]
I recommend adding 2.7.5 to your tox.ini

**Comment by davidism** [comment]
It wouldn't do any good, most people don't have 2.7.5 installed, adding an entry to tox wouldn't solve that. Additionally we don't have unlimited resources to check every patch version of every Python on CI. At some point we have to assume people install patches.

**Comment by golf-player** [comment]
True for the most part, but imo the versions that ship with CentOS and are
in epel are good to have just because they're the ones commonly in usage.

**Comment by asottile** [comment]
#1545 has a fix, wasn't too difficult.  tox both doesn't support 3-number versions and doens't help you install that version of python.

Neither travis-ci nor azure pipelines ship with that old of a python version either -- we would have to do some complex docker trickery to end up with that version which simply isn't worth it for a soon-dead version of python

**Comment by golf-player** [comment]
Makes sense. Forget I asked.

**Comment by asottile** [comment]
nah it's fine <3 -- good thought on putting some testing in place to prevent regressions, it's just prohibitively difficult in this situation

**Comment by davidism** [comment]
Released 0.15.4

**Comment by golf-player** [comment]
That was fast! thanks!
